File: app/src/main/java/md/fusionworks/lifehack/data/persistence/TaxiDataStore.java
package md.fusionworks.lifehack.data.persistence;

import io.realm.Realm;
import java.util.Date;
import md.fusionworks.lifehack.data.persistence.model.TaxiPhoneNumber;

/**
 * Created by ungvas on 2/18/16.
 */
public class TaxiDataStore {

  private Realm realm;

  public TaxiDataStore(Realm realm) {
    this.realm = realm;
  }

  public void add(int[] phoneNumberList) {
    realm.executeTransaction(realm -> {
      for (int i = 0; i < phoneNumberList.length; i++) {
        TaxiPhoneNumber taxiPhoneNumber = realm.createObject(TaxiPhoneNumber.class);
        taxiPhoneNumber.setPhoneNumber(phoneNumberList[i]);
        taxiPhoneNumber.setLastCallDate(null);
      }
    });
  }

  public void updateLastUsedDate(int phoneNumber) {
    realm.executeTransaction(realm -> {
      TaxiPhoneNumber taxiPhoneNumber =
          realm.where(TaxiPhoneNumber.class).equalTo("phoneNumber", phoneNumber).findFirst();
      if (taxiPhoneNumber.isLoaded()) taxiPhoneNumber.setLastCallDate(new Date());
    });
  }
}

File: app/src/main/java/md/fusionworks/lifehack/taxi/persistence/TaxiDataStore.kt
package md.fusionworks.lifehack.taxi.persistence

import io.realm.Realm
import md.fusionworks.lifehack.di.PerActivity
import java.util.*
import javax.inject.Inject

/**
 * Created by ungvas on 2/18/16.
 */
@PerActivity
class TaxiDataStore
@Inject constructor(private val realm: Realm) {

  fun add(phoneNumberList: IntArray) = realm.executeTransaction { realm ->
    for (i in phoneNumberList.indices) {
      val taxiPhoneNumber = realm.createObject(TaxiPhoneNumber::class.java)
      taxiPhoneNumber.phoneNumber = phoneNumberList[i]
      taxiPhoneNumber.lastCallDate = null
    }
  }

  fun updateLastUsedDate(phoneNumber: Int) = realm.executeTransaction { realm ->
    val taxiPhoneNumber = realm.where(TaxiPhoneNumber::class.java).equalTo("phoneNumber",
        phoneNumber).findFirst()
    if (taxiPhoneNumber.isLoaded) taxiPhoneNumber.lastCallDate = Date()
  }
}
